### PR TITLE
Remove need to have ConjureCollections

### PR DESCRIPTION
**Pull request by markelliot**

### CODE REVIEW

**Comment by markelliot** [comment]
I was looking at something else and noticed `ConjureCollections`, which we added to deal with iterables, but really shouldn't be required, so I removed it.

File: conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureCollections.java [hunk 1]
```diff
@@ -1,41 +0,0 @@
...
-public final class ConjureCollections {
-
-    private ConjureCollections() {
-        // cannot instantiate
-    }
-
-    @SuppressWarnings("unchecked")
-    public static <T> void addAll(Collection<T> addTo, Iterable<? extends T> elementsToAdd) {
-        if (elementsToAdd instanceof Collection) {
-            addTo.addAll((Collection<T>) elementsToAdd);
```
Comment (iamdanfox) [inline conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureCollections.java:34]: I think the point of having this method was that it's more performant to do `addAll` where possible...

Comment (markelliot) [reply conjure-lib/src/main/java/com/palantir/conjure/java/lib/internal/ConjureCollections.java:34]: Depends on the data structure, I guess:

LinkedHashSet:
```java
   public boolean addAll(Collection<? extends E> c) {
        boolean modified = false;
        for (E e : c)
            if (add(e))
                modified = true;
        return modified;
    }
```

ArrayList:
```java
    public boolean addAll(Collection<? extends E> c) {
        Object[] a = c.toArray();
        int numNew = a.length;
        ensureCapacityInternal(size + numNew);  // Increments modCount
        System.arraycopy(a, 0, elementData, size, numNew);
        size += numNew;
        return numNew != 0;
    }
```

**Comment by markelliot** [comment]
@iamdanfox one flag on this one, it probably ought to be documented in ConjureCollections why we do what we do.

**Comment by iamdanfox** [comment]
Have opened a PR here: https://github.com/palantir/conjure-java/pull/24

**Pull request closed without merging**